osu.Game.Rulesets.Osu/Mods/OsuModComboScaling.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Localisation;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.UI;
+using osu.Game.Rulesets.Osu.UI.Cursor;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Scoring;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModComboScaling : Mod, IApplicableToDifficulty, IApplicableToScoreProcessor, IUpdatableByPlayfield
+    {
+        public override string Name => "Combo Scaling";
+        public override string Acronym => "CS";
+        public override LocalisableString Description => "Circle size adjusts dynamically based on your combo.";
+        public override ModType Type => ModType.Fun;
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Concat(new[]
+        {
+            typeof(OsuModAutoplay),
+            typeof(ModCinema),
+            typeof(ModEasy),
```

Comment:
i think difficulty adjust, depth, grow, and deflate should also be incompatible?